eth-contracts/contracts/service/storage/ServiceProviderStorage.sol

```diff
@@ -147,6 +147,32 @@ contract ServiceProviderStorage is RegistryContract {
         serviceProviderInfo[_serviceType][spID].delegateOwnerWallet = _updatedDelegateOwnerWallet;
     }
 
+    function updateEndpoint(
+        address _owner,
+        bytes32 _serviceType,
+        string calldata _oldEndpoint,
+        string calldata _newEndpoint
+    ) external onlyRegistrant(CALLER_REGISTRY_KEY) returns (uint spID)
+    {
+        uint spId = this.getServiceProviderIdFromEndpoint(_oldEndpoint);
```

Comment:
Look into not using _oldEndpoint. This should come from the ServiceProvider struct